main.tf

```diff
@@ -2,158 +2,110 @@ data "azurerm_resource_group" "main" {
-        subnet_id    = var.ingress_application_gateway_subnet_id
-      }
+    content {
+      log_analytics_workspace_id = azurerm_log_analytics_workspace.main[0].id
     }
   }
 
-  role_based_access_control {
-    enabled = var.enable_role_based_access_control
-
-    dynamic "azure_active_directory" {
-      for_each = var.enable_role_based_access_control && var.rbac_aad_managed ? ["rbac"] : []
-      content {
-        managed                = true
-        admin_group_object_ids = var.rbac_aad_admin_group_object_ids
-      }
+  dynamic "ingress_application_gateway" {
+    for_each = var.ingress_application_gateway_enabled ? { "loop" = "once" } : {}
+    content {
+      gateway_id   = var.gateway_id
+      gateway_name = var.gateway_name
+      subnet_cidr  = var.subnet_cidr
+      subnet_id    = var.subnet_id
     }
+  }
 
-    dynamic "azure_active_directory" {
-      for_each = var.enable_role_based_access_control && !var.rbac_aad_managed ? ["rbac"] : []
-      content {
-        managed           = false
-        client_app_id     = var.rbac_aad_client_app_id
-        server_app_id     = var.rbac_aad_server_app_id
-        server_app_secret = var.rbac_aad_server_app_secret
-      }
-    }
+  azure_active_directory_role_based_access_control {
+    managed                = var.rbac_aad_managed
+    admin_group_object_ids = var.rbac_aad_managed ? var.rbac_aad_admin_group_object_ids : null
+    azure_rbac_enabled     = var.rbac_aad_managed ? var.enable_role_based_access_control : null
+    client_app_id          = !var.rbac_aad_managed ? var.rbac_aad_client_app_id : null
+    server_app_id          = !var.rbac_aad_managed ? var.rbac_aad_server_app_id : null
+    server_app_secret      = !var.rbac_aad_managed ? var.rbac_aad_server_app_secret : null
   }
 
   network_profile {
     network_plugin     = var.network_plugin
     network_policy     = var.network_policy
-    dns_service_ip     = var.net_profile_dns_service_ip
-    docker_bridge_cidr = var.net_profile_docker_bridge_cidr
-    outbound_type      = var.net_profile_outbound_type
-    pod_cidr           = var.net_profile_pod_cidr
-    service_cidr       = var.net_profile_service_cidr
+    dns_service_ip     = var.dns_service_ip
+    docker_bridge_cidr = var.docker_bridge_cidr
+    outbound_type      = var.outbound_type
+    pod_cidr           = var.network_plugin != "azure" ? var.pod_cidr : null
+    service_cidr       = var.service_cidr
   }
 
   tags = var.tags
 }
 
-
+# Log Analytics
 resource "azurerm_log_analytics_workspace" "main" {
-  count               = var.enable_log_analytics_workspace ? 1 : 0
-  name                = var.cluster_log_analytics_workspace_name == null ? "${var.prefix}-workspace" : var.cluster_log_analytics_workspace_name
+  count               = var.log_analytics_workspace_enabled ? 1 : 0
+  name                = var.log_analytics_workspace_name == null ? "${try(var.prefix, var.cluster_name)}-workspace" : var.log_analytics_workspace_name
```

Comment:
When I was testing this, I ran into an issue here:
```
│ Error: Invalid template interpolation value
│
│   on .terraform/modules/aks/main.tf line 108, in resource "azurerm_log_analytics_workspace" "main":
│  108:   name                = var.log_analytics_workspace_name == null ? "${try(var.prefix, var.cluster_name)}-workspace" : var.log_analytics_workspace_name
│     ├────────────────
│     │ var.cluster_name is "test-aks"
│     │ var.prefix is null
│
│ The expression result is null. Cannot include a null value in a string template.
```
What fixed it for me--and what I'd recommend--is to use `coalesce(var.cluster_name, var.prefix)` instead of `try()`.